vaadin-grid-flow/src/test/java/com/vaadin/flow/component/grid/GridDndMobilePolyfillTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.grid;
+
+import java.util.Properties;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.grid.dnd.GridDropMode;
+import com.vaadin.flow.server.DefaultDeploymentConfiguration;
+import com.vaadin.flow.server.VaadinService;
+import com.vaadin.flow.server.VaadinServlet;
+import com.vaadin.flow.server.VaadinSession;
+import com.vaadin.flow.server.WebBrowser;
+
+public class GridDndMobilePolyfillTest {
+
+    private Grid<String> grid;
+    private UI ui;
+
+    @Before
+    public void setup() {
+        DefaultDeploymentConfiguration configuration = new DefaultDeploymentConfiguration(
+                VaadinServlet.class, new Properties()) {
+            @Override
+            public boolean isCompatibilityMode() {
+                return false;
+            }
+        };
+
+        VaadinService service = Mockito.mock(VaadinService.class);
+        Mockito.when(service.resolveResource(Mockito.anyString(),
+                Mockito.any(WebBrowser.class))).thenReturn("");
+
+        VaadinSession session = Mockito.mock(VaadinSession.class);
+        Mockito.when(session.getConfiguration()).thenReturn(configuration);
+        Mockito.when(session.getService()).thenReturn(service);
+        Mockito.when(session.hasLock()).thenReturn(true);
+        ui = new UI() {
+            @Override
+            public VaadinSession getSession() {
+                return session;
+            }
+        };
+        ui.getInternals().setSession(session);
+
+        grid = new Grid<>();
+        ui.add(grid);
+    }
+
+    @Test
+    public void gridDnd_setDropMode_mobilePolyfillShouldBeAdded() {
+        grid.setDropMode(GridDropMode.ON_GRID);
+        Assert.assertEquals(
+                "Grid::setDropMode should add DnD mobile polyfill script to the UI.",
+                1, ui.getInternals().dumpPendingJavaScriptInvocations().size());
+    }
+
+    @Test
+    public void gridDnd_setRowsDraggable_mobilePolyfillShouldBeAdded() {
+        grid.setRowsDraggable(true);
```

Comment:
Same: Since there is no guarantee that the setup phase has not added already a pending JS invocation, you should do a dump before calling the grid method to make sure the added invocation is not from somewhere else.

Reply:
Done